shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/messaging/AggregatingMetaStoreEventReader.java

```diff
@@ -48,11 +48,9 @@
       long maxExecTime = windowUnits.toMillis(window);
       long startTime = System.currentTimeMillis();
       while (startTime + maxExecTime > System.currentTimeMillis()) {
-        // TODO: since delegate.next() effectively blocks until some messages arrive, the below
```

Comment:
I think this TODO is still true as an issue and so we should probably keep the comment, even if not as a TODO but more as a warning.

Reply:
Yes this is still _kind of_ true. Previously, `SqsMessageReader.next()` would block until messages are returned:

`while (records == null || !records.hasNext()) { records = consumer.receiveMessage(...)... }`

now, however, it will only block for `waitTimeSeconds`, the amount of time to wait for messages to arrive, as I've removed the loop to allow whoever uses this library to implement this as they like. Happy to add this back in as a warning if you think it still warrants it?

Reply:
Ah, OK, so that's very hard to see since the change is in what is showing up as a new file in the other PR. Anyway, from what I can see that's done by setting the `withWaitTimeSeconds` on the `ReceiveMessageRequest` in `SqsMessageReader` which is in the other PR. I know we had some long discussions about how this all works a few months ago, when my brain has some capacity I'd like to think if there are any edge cases this change will have an impact on. Does the message reader return an empty list if the timeout occurs? So it will keep looping since an empty message list is returned but it won't block? 

As for the TODO comment, I guess the change means there could still be this problem if you provide a weird combination of values - like a `waitTimeSeconds` bigger than the `maxExecTime` but that's not our fault so we don't need to document it...

Reply:
Yes, from the aws docs [here]:(https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/sqs/model/ReceiveMessageRequest.html#getWaitTimeSeconds--) 

> If no messages are available and the wait time expires, the call returns successfully with an empty list of messages.

So if the `ReceiveMessageRequest` times out it will return an empty list, which will be returned as `Optional.empty()` by `SqsMessageReader`. 

`AggregatingMetaStoreEventReader` will then continue the loop to read (until `startTime + maxExecTime > System.currentTimeMillis()). So the block will only happen for the duration of `waitTimeSeconds` while waiting to receive the message.